pyClient/commands/utils.py

```diff
@@ -244,29 +243,26 @@ def find_pub_address_file(base_file: str) -> str:
     raise ClickException(f"No public key file {pub_addr_file} or {base_file}")
 
 
-def create_zeth_client(ctx: Context) -> ZethClient:
+def create_mixer_client(ctx: ClientConfig) -> MixerClient:
     """
-    Create a ZethClient for an existing deployment.
+    Create a MixerClient for an existing deployment.
     """
     web3 = open_web3_from_ctx(ctx)
     mixer_desc = load_mixer_description_from_ctx(ctx)
     mixer_instance = mixer_desc.mixer.instantiate(web3)
-    prover_client = ctx.obj["PROVER_CLIENT"]
-    zksnark = ctx.obj["ZKSNARK"]
-    return ZethClient.open(web3, prover_client, mixer_instance, zksnark)
+    return MixerClient.open(web3, ctx.prover_server_endpoint, mixer_instance)
 
 
 def create_zeth_client_and_mixer_desc(
-        ctx: Context) -> Tuple[ZethClient, MixerDescription]:
+        ctx: ClientConfig) -> Tuple[MixerClient, MixerDescription]:
     """
-    Create a ZethClient and MixerDescription object, for an existing deployment.
+    Create a MixerClient and MixerDescription object, for an existing deployment.
     """
     web3 = open_web3_from_ctx(ctx)
     mixer_desc = load_mixer_description_from_ctx(ctx)
     mixer_instance = mixer_desc.mixer.instantiate(web3)
-    prover_client = ctx.obj["PROVER_CLIENT"]
-    zksnark = ctx.obj["ZKSNARK"]
-    zeth_client = ZethClient.open(web3, prover_client, mixer_instance, zksnark)
+    zeth_client = MixerClient.open(
```

Comment:
I felt this was a bit more accurate, since it mainly prepares the mix parameters and sends them to the mixer.  To me, a full "zeth client" would also include the wallet and the logic to handle the emitted events etc, so when I did the small refactor of the wallet ownership. The `Wallet` handles the management of zeth state data (merkle tree) and notes, and the `MixerClient` deals specifically with interacting with the mixer contract.

To be clear, I think there is some further refactoring to be done to add / remove some functionality from here, but hopefully the above explains my thinking.

If you prefer to change the name back it's not a big deal.

Reply:
I see, that makes sense indeed. Thanks!